src/plugins/badge.coffee

```diff
@@ -608,4 +610,45 @@ class Badge
 
     "#{settings.url.application_base}/badge/shop_reviews?#{serialized_params}"
 
+  _initHideOnScroll: (badge) =>
+    past_scroll = window.parent.pageYOffset
+    return unless past_scroll? # pageYOffset is not supported for IE < 9
+
+    @_attachEvent @parent_doc, 'scroll', (_event) =>
+      current_scroll = window.parent.pageYOffset
+
+      return if current_scroll == past_scroll # Ignore horizontal scrolling
+      return if @_outOfBound(current_scroll) # Ignore OSX bounce
+
+      scroll_diff = current_scroll - past_scroll
+      return if Math.abs(scroll_diff) <= 5
+
+      badge.style.display = if scroll_diff > 0 then 'none' else 'block'
+
+      past_scroll = current_scroll
+
+  _outOfBound: (current_scroll) ->
+    past_top = current_scroll < 0
+
+    document_height = Math.max @parent_doc.body.scrollHeight, @parent_doc.documentElement.scrollHeight,
+                               @parent_doc.body.offsetHeight, @parent_doc.documentElement.offsetHeight,
+                               @parent_doc.body.clientHeight, @parent_doc.documentElement.clientHeight
+
+    past_bottom = current_scroll + @_vieport().height > document_height
+
+    past_top || past_bottom
+
+  _smallDevice: -> @_vieport().width <= 768
+
+  _vieport: ->
+    e = window.parent
+    a = 'inner'
+
+    if !('innerWidth' of e)
```

Comment:
Could use `unless`, and `hasOwnProperty` seems a better fit as we don't want to check for an inherited property.
`unless e.hasOwnProperty('innerWidth')`

----------
And I suggest a different approach, which I think is simpler and more readable:
```
_vieport: ->
    if window.parent.hasOwnProperty('innerWidth')
      width: window.parent.innerWidth
      heigth: window.parent.innerWidth
    else
      width: (@parent_doc.documentElement or @parent_doc.body).clientWidth
      height: (@parent_doc.documentElement or @parent_doc.body).clientHeight
```